dbt_subprojects/solana/macros/_project/oneinch/oneinch_fusion_transfers_macro.sql

```diff
@@ -0,0 +1,62 @@
+{% macro oneinch_fusion_transfers_macro(direction) %} -- src/dst
+
+
+select 
+        tx_id
+        , block_time
+        , block_slot
+        , taker as resolver -- Check if this is correct logic
+        , maker as user -- Check
+        , order_hash
+        , tx_success
+        , call_trace_address
+        , token_mint_address as {{ direction }}_token_address
+        , symbol as {{ direction }}_token_symbol
+        , amount as {{ direction }}_token_amount
+        , amount_usd as {{ direction }}_token_amount_usd
+        -- , CALLTRACEADDRESs
+
+    -- , outer_executing_account
+    -- , tx_id
+    -- , outer_instruction_index
+    -- , block_time
+    -- , block_slot
+    -- , min_by(token_mint_address, inner_instruction_index) as {{ direction }}_token_mint
+    -- , min_by(amount_usd, inner_instruction_index) as {{ direction }}_amount_usd
+    -- , max_by(token_mint_address, inner_instruction_index) as dst_token_mint
+    -- , max_by(amount_usd, inner_instruction_index) as dst_amount_usd
+    -- , coalesce(max(amount_usd), 0) as max_amount_usd
+    -- , coalesce(min(amount_usd), 0) as min_amount_usd
+from (
+    select 
+        tx_id
+        , block_time
+        , block_slot
+        , taker
+        , maker
+        , order_hash
+        , {{ direction }}_mint as token_mint_address
+        , tx_success
+        , array[outer_instruction_index, inner_instruction_index] as call_trace_address
+    from {{ ref('oneinch_solana_fusion_calls') }}
+    where 
+        instruction_type = 'fill'
+        {% if is_incremental() %}
+            and {{ incremental_predicate('block_time') }}
+        {% else %}
+            and block_time >= date('{{ oneinch_cfg_macro("project_start_date") }}')
+        {% endif %}
+) qf
+left join {{ ref('tokens_solana_transfers') }} using(tx_id, token_mint_address, block_time, block_slot)
```

Comment:
may be worth monitoring performance on this join and if adding incremental predicates will help as data grows

Reply:
thx for noticing! missed this

Reply:
fixed

Reply:
fixed